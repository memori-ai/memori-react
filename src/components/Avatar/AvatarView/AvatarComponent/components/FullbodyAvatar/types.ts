export enum AnimationState {
  LOADING = 'LOADING',
  EMOTION = 'EMOTION',
  IDLE = 'IDLE',
}

export interface AnimationConfig {
  fadeInDuration: number;
  fadeOutDuration: number;
  idleCount: number;
  timeScale: number;
}

export interface FullbodyAvatarProps {
  url: string;
  sex: 'MALE' | 'FEMALE';
  onLoaded?: () => void;
  currentBaseAction: {
    action: string;
    weight: number;
  };
  timeScale: number;
  onCameraZChange: (value: number) => void;
  eyeBlink?: boolean;
  avatarDepth?: number;
  avatarHeight?: number;
  stopProcessing: () => void;
  resetVisemeQueue: () => void;
  updateCurrentViseme: (
    currentTime: number
  ) => { name: string; weight: number } | null;
  smoothMorphTarget?: boolean;
  morphTargetSmoothing?: number;
  morphTargetInfluences: Record<string, number>;
  setMorphTargetDictionary: (
    morphTargetDictionary: Record<string, number>
  ) => void;
  setMorphTargetInfluences: (
    morphTargetInfluences: Record<string, number>
  ) => void;
  emotionMorphTargets: Record<string, number>;
  halfBody: boolean;
}
