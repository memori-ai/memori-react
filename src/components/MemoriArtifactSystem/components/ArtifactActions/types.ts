/**
 * TypeScript interfaces for artifact copy system components
 */

import React from 'react';

export interface CopyFormat {
  id: string;
  label: string;
  description?: string;
  icon?: React.ReactNode | undefined;
  mimeType: string;
  action: 'copy' | 'pdf' | 'print' | 'download' | 'link';
  isAsync?: boolean;
}

export interface CopyMenuItemProps {
  format: CopyFormat;
  onClick: (format: CopyFormat) => void;
  loading?: boolean;
  disabled?: boolean;
}

// CopyDropdownProps removed - using headless-ui Menu directly

export interface CopyButtonWithDropdownProps {
  artifact: {
    content: string;
    mimeType: string;
    title?: string;
  };
  onCopy?: () => void;
  onDownload?: () => void;
  onPrint?: () => void;
  onOpenExternal?: () => void;
  loading?: boolean;
  className?: string;
  disabled?: boolean;
}

export interface CopyState {
  isOpen: boolean;
  loading: boolean;
  success: boolean;
  error: string | null;
  activeFormat: string | null;
}

export interface UseCopyArtifactReturn {
  copyState: CopyState;
  formats: CopyFormat[];
  handleCopy: (format: CopyFormat) => Promise<void>;
  toggleDropdown: () => void;
  closeDropdown: () => void;
  resetState: () => void;
  handleCopyClick: () => Promise<void>;
}

export interface PDFExportOptions {
  title?: string;
  author?: string;
  margin?: {
    top: string;
    right: string;
    bottom: string;
    left: string;
  };
  fontSize?: string;
  fontFamily?: string;
  lineHeight?: string;
  color?: string;
  backgroundColor?: string;
}
