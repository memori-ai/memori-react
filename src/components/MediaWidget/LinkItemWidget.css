.memori-link-items--grid {
    display: grid;
    padding: 1rem;
    grid-auto-flow: dense;
    grid-gap: 1rem;
    grid-template-columns: repeat(auto-fill, minmax(150px, 1fr));
}

.memori-link-item--card {
    height: 100%;
    margin-bottom: 1rem;
}

.memori-link-item--card .memori-card--cover {
    height: auto;
    text-align: center;
}

.memori-link-item--card .memori-card--cover img {
    min-width: 100%;
    min-height: 100%;
    object-fit: contain;
}

a.memori-link-item--link {
    text-decoration: none;
}

.memori-link-item--card-cover-icon {
    padding: 1rem;
}

.memori-link-item--icon {
    width: 50%;
    height: 50%;
}