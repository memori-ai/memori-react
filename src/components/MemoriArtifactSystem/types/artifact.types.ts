export interface ArtifactData {
  id: string;
  content: string;
  mimeType: string;
  title: string;
  timestamp: Date;
  size: number;
}

export interface ArtifactSystemState {
  currentArtifact: ArtifactData | null;
  isDrawerOpen: boolean;
  isFullscreen: boolean;
}

export type ArtifactTab = 'code' | 'preview';