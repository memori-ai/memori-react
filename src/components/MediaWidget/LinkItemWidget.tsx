import { Medium } from '@memori.ai/memori-api-client/dist/types';
import React, { useState, useEffect } from 'react';
import Link from '../icons/Link';
import cx from 'classnames';
import { ellipsis } from 'ellipsed';
import Card from '../ui/Card';
import { Transition } from '@headlessui/react';

import './LinkItemWidget.css';

export declare type ILinkPreviewInfo = {
  title?: string | undefined;
  siteName?: string | undefined;
  description?: string | undefined;
  mediaType?: string | undefined;
  image?: string | undefined;
  imageWidth?: number | undefined;
  imageHeight?: number | undefined;
  favicon?: string | undefined;
  images?: string[] | undefined;
  video?: string | undefined;
  videos?: string[] | undefined;
};

export interface Props {
  items: Medium[];
  baseUrl?: string;
  descriptionOneLine?: boolean;
  onLinkPreviewInfo?: (linkPreviewInfo: ILinkPreviewInfo) => void;
}

const getSiteInfo = async (url: string, baseUrl?: string) => {
  try {
    const data = await fetch(
      `${baseUrl ||
        'https://app.twincreator.com'}/api/linkpreview/${encodeURIComponent(
        url
      )}`
    );
    const siteInfo: ILinkPreviewInfo = await data.json();
    return siteInfo;
  } catch (err) {
    console.error('getSiteInfo', err);
    return null;
  }
};

export const RenderLinkItem = ({
  item,
  baseUrl,
  onLinkPreviewInfo,
  descriptionOneLine = false,
}: Omit<Props, 'items'> & {
  item: Medium;
}) => {
  const [link, setLink] = useState<
    (ILinkPreviewInfo & { urlKey: string }) | null
  >(null);
  const normURL =
    item.url?.startsWith('http') || !item.url || item.url?.length === 0
      ? item.url
      : `https://${item.url}`;

  useEffect(() => {
    let normURL =
      item.url?.startsWith('http') || !item.url || item.url?.length === 0
        ? item.url
        : `https://${item.url}`;
    if (normURL && normURL !== link?.urlKey) {
      getSiteInfo(normURL, baseUrl).then(siteInfo => {
        setLink(
          siteInfo
            ? ({ ...siteInfo, urlKey: normURL } as ILinkPreviewInfo & {
                urlKey: string;
              })
            : null
        );

        if (onLinkPreviewInfo && siteInfo) onLinkPreviewInfo(siteInfo);
      });
    }
  }, [item?.url, link, baseUrl, onLinkPreviewInfo]);

  const title = item.title && item.title.length > 0 ? item.title : link?.title;
  const description = link?.description;
  const video = link?.video;
  const image = link?.image ?? link?.images?.[0];

  useEffect(() => {
    setTimeout(() => {
      ellipsis('.memori-link-item--card .memori-card--description', 3, {
        responsive: true,
      });
    }, 300);
  }, [link?.description, item.mediumID]);

  return (
    <a
      href={normURL}
      target="_blank"
      rel="noopener noreferrer"
      className="memori-link-item--link"
    >
      <Card
        hoverable
        className={cx('memori-link-item--card', {
          'memori-link-item--card-description-oneline': descriptionOneLine,
          'memori-link-item--card-has-image': image,
          'memori-link-item--card-has-video': video,
        })}
        cover={
          video ? (
            <iframe
              width="100%"
              height="100%"
              src={video}
              title="Video player"
              frameBorder="0"
              allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
              allowFullScreen
            />
          ) : image ? (
            <img
              className="memori-link-item--card-cover-img"
              src={image}
              alt={item.title}
            />
          ) : (
            <div className="memori-link-item--card-cover-icon">
              <Link className="memori-link-item--icon" />
            </div>
          )
        }
        title={title}
        description={description}
      />
    </a>
  );
};

const LinkItemWidget: React.FC<Props> = ({
  items,
  baseUrl,
  descriptionOneLine = false,
  onLinkPreviewInfo,
}: Props) => {
  return (
    <Transition appear show as="div" className="memori-link-items">
      <div className="memori-link-items--grid">
        {items.map((item, index) => (
          <Transition.Child
            as="div"
            className="memori-link-item"
            key={item.mediumID}
            enter={`ease-out duration-500 delay-${index * 100}`}
            enterFrom="opacity-0 scale-95"
            enterTo="opacity-1 scale-100"
            leave="ease-in duration-300"
            leaveFrom="opacity-1 scale-100"
            leaveTo="opacity-0 scale-95"
          >
            <RenderLinkItem
              item={item}
              baseUrl={baseUrl}
              descriptionOneLine={descriptionOneLine}
              onLinkPreviewInfo={onLinkPreviewInfo}
            />
          </Transition.Child>
        ))}
      </div>
    </Transition>
  );
};

export default LinkItemWidget;
