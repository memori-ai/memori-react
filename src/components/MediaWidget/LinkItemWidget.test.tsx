import React from 'react';
import { Medium } from '@memori.ai/memori-api-client/dist/types';
import { render } from '@testing-library/react';
import LinkItemWidget from './LinkItemWidget';

const linkMedium: Medium = {
  mediumID: '95226d7e-7bae-465e-8b80-995587bb5971',
  mimeType: 'text/html',
  title: 'Memori Srl',
  url: 'https://memori.ai/en',
};

it('renders LinkItemWidget unchanged', () => {
  const { container } = render(<LinkItemWidget items={[linkMedium]} />);
  expect(container).toMatchSnapshot();
});

it('renders LinkItemWidget unchanged with no media', () => {
  const { container } = render(<LinkItemWidget items={[]} />);
  expect(container).toMatchSnapshot();
});

it('renders LinkItemWidget unchanged with baseUrl', () => {
  const { container } = render(<LinkItemWidget items={[linkMedium]} />);
  expect(container).toMatchSnapshot();
});

it('renders LinkItemWidget unchanged with description oneline', () => {
  const { container } = render(
    <LinkItemWidget items={[linkMedium]} descriptionOneLine />
  );
  expect(container).toMatchSnapshot();
});
