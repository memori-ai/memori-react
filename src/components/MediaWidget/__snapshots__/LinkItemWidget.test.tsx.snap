// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LinkItemWidget unchanged 1`] = `
<div>
  <div
    class="memori-link-items"
  >
    <div
      class="memori-link-items--grid"
    >
      <div
        class="memori-link-item ease-out duration-500 delay-0 opacity-0 scale-95"
      >
        <a
          class="memori-link-item--link"
          href="https://memori.ai/en"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="memori-card memori-link-item--card memori-card--with-cover memori-card--hoverable"
          >
            <div
              class="memori-spin"
            >
              <div
                class="memori-card--cover"
              >
                <div
                  class="memori-link-item--card-cover-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="memori-link-item--icon"
                    focusable="false"
                    role="img"
                    viewBox="0 0 1024 1024"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M574 665.4a8.03 8.03 0 0 0-11.3 0L446.5 781.6c-53.8 53.8-144.6 59.5-204 0-59.5-59.5-53.8-150.2 0-204l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3l-39.8-39.8a8.03 8.03 0 0 0-11.3 0L191.4 526.5c-84.6 84.6-84.6 221.5 0 306s221.5 84.6 306 0l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3L574 665.4zm258.6-474c-84.6-84.6-221.5-84.6-306 0L410.3 307.6a8.03 8.03 0 0 0 0 11.3l39.7 39.7c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c53.8-53.8 144.6-59.5 204 0 59.5 59.5 53.8 150.2 0 204L665.3 562.6a8.03 8.03 0 0 0 0 11.3l39.8 39.8c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c84.5-84.6 84.5-221.5 0-306.1zM610.1 372.3a8.03 8.03 0 0 0-11.3 0L372.3 598.7a8.03 8.03 0 0 0 0 11.3l39.6 39.6c3.1 3.1 8.2 3.1 11.3 0l226.4-226.4c3.1-3.1 3.1-8.2 0-11.3l-39.5-39.6z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="memori-card--content"
              >
                <h3
                  class="memori-card--title"
                >
                  Memori Srl
                </h3>
                <div
                  class="memori-card--children"
                />
              </div>
              <div
                class="memori-spin--spinner"
              >
                <svg
                  aria-hidden="true"
                  class="memori-loading-icon"
                  focusable="false"
                  role="img"
                  viewBox="0 0 1024 1024"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders LinkItemWidget unchanged with baseUrl 1`] = `
<div>
  <div
    class="memori-link-items"
  >
    <div
      class="memori-link-items--grid"
    >
      <div
        class="memori-link-item ease-out duration-500 delay-0 opacity-0 scale-95"
      >
        <a
          class="memori-link-item--link"
          href="https://memori.ai/en"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="memori-card memori-link-item--card memori-card--with-cover memori-card--hoverable"
          >
            <div
              class="memori-spin"
            >
              <div
                class="memori-card--cover"
              >
                <div
                  class="memori-link-item--card-cover-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="memori-link-item--icon"
                    focusable="false"
                    role="img"
                    viewBox="0 0 1024 1024"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M574 665.4a8.03 8.03 0 0 0-11.3 0L446.5 781.6c-53.8 53.8-144.6 59.5-204 0-59.5-59.5-53.8-150.2 0-204l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3l-39.8-39.8a8.03 8.03 0 0 0-11.3 0L191.4 526.5c-84.6 84.6-84.6 221.5 0 306s221.5 84.6 306 0l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3L574 665.4zm258.6-474c-84.6-84.6-221.5-84.6-306 0L410.3 307.6a8.03 8.03 0 0 0 0 11.3l39.7 39.7c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c53.8-53.8 144.6-59.5 204 0 59.5 59.5 53.8 150.2 0 204L665.3 562.6a8.03 8.03 0 0 0 0 11.3l39.8 39.8c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c84.5-84.6 84.5-221.5 0-306.1zM610.1 372.3a8.03 8.03 0 0 0-11.3 0L372.3 598.7a8.03 8.03 0 0 0 0 11.3l39.6 39.6c3.1 3.1 8.2 3.1 11.3 0l226.4-226.4c3.1-3.1 3.1-8.2 0-11.3l-39.5-39.6z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="memori-card--content"
              >
                <h3
                  class="memori-card--title"
                >
                  Memori Srl
                </h3>
                <div
                  class="memori-card--children"
                />
              </div>
              <div
                class="memori-spin--spinner"
              >
                <svg
                  aria-hidden="true"
                  class="memori-loading-icon"
                  focusable="false"
                  role="img"
                  viewBox="0 0 1024 1024"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders LinkItemWidget unchanged with description oneline 1`] = `
<div>
  <div
    class="memori-link-items"
  >
    <div
      class="memori-link-items--grid"
    >
      <div
        class="memori-link-item ease-out duration-500 delay-0 opacity-0 scale-95"
      >
        <a
          class="memori-link-item--link"
          href="https://memori.ai/en"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="memori-card memori-link-item--card memori-link-item--card-description-oneline memori-card--with-cover memori-card--hoverable"
          >
            <div
              class="memori-spin"
            >
              <div
                class="memori-card--cover"
              >
                <div
                  class="memori-link-item--card-cover-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="memori-link-item--icon"
                    focusable="false"
                    role="img"
                    viewBox="0 0 1024 1024"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M574 665.4a8.03 8.03 0 0 0-11.3 0L446.5 781.6c-53.8 53.8-144.6 59.5-204 0-59.5-59.5-53.8-150.2 0-204l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3l-39.8-39.8a8.03 8.03 0 0 0-11.3 0L191.4 526.5c-84.6 84.6-84.6 221.5 0 306s221.5 84.6 306 0l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3L574 665.4zm258.6-474c-84.6-84.6-221.5-84.6-306 0L410.3 307.6a8.03 8.03 0 0 0 0 11.3l39.7 39.7c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c53.8-53.8 144.6-59.5 204 0 59.5 59.5 53.8 150.2 0 204L665.3 562.6a8.03 8.03 0 0 0 0 11.3l39.8 39.8c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c84.5-84.6 84.5-221.5 0-306.1zM610.1 372.3a8.03 8.03 0 0 0-11.3 0L372.3 598.7a8.03 8.03 0 0 0 0 11.3l39.6 39.6c3.1 3.1 8.2 3.1 11.3 0l226.4-226.4c3.1-3.1 3.1-8.2 0-11.3l-39.5-39.6z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="memori-card--content"
              >
                <h3
                  class="memori-card--title"
                >
                  Memori Srl
                </h3>
                <div
                  class="memori-card--children"
                />
              </div>
              <div
                class="memori-spin--spinner"
              >
                <svg
                  aria-hidden="true"
                  class="memori-loading-icon"
                  focusable="false"
                  role="img"
                  viewBox="0 0 1024 1024"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders LinkItemWidget unchanged with no media 1`] = `
<div>
  <div
    class="memori-link-items"
  >
    <div
      class="memori-link-items--grid"
    />
  </div>
</div>
`;
